### ISSUE TITLE
Is Swap (1,998) too easy?

### ISSUE THREAD

**Issue by robbiespeed** [krausest/js-framework-benchmark#1270]
At least in the case of Solid it's using a method of diffing that starts from the outer ends of an array and narrows in on differing elements as it gets closer to center, since both 2 and 999 are near both ends the elements are immediately swapped. I imagine other frameworks are using this same method as well.

It could be better for swap to have a defined set of swap locations to achieve a more true test. ex: `[[1, 998], [2, 3], [5, 248], [249, 250], [251, 997]]`

**Comment by localvoid** [comment]
> I imagine other frameworks are using this same method as well.

No, ivi [finds the minimum number of DOM ops](https://github.com/localvoid/ivi/blob/43cdf6f747dc782883ca73bdb5b4e21fa9c27655/packages/ivi/src/client/core.ts#L1280-L1329) and doesn't use "prefix/suffix move" optimization.

> It could be better for swap to have a defined set of swap locations to achieve a more true test

It won't make any difference for libraries that use "prefix/suffix move" optimization.

**Comment by robbiespeed** [comment]
> No, ivi [finds the minimum number of DOM ops](https://github.com/localvoid/ivi/blob/43cdf6f747dc782883ca73bdb5b4e21fa9c27655/packages/ivi/src/client/core.ts#L1280-L1329) and doesn't use "prefix/suffix move" optimization.

I meant there must be other frameworks using the optimization not only Solid, not that all other frameworks use it.

> It won't make any difference for libraries that use "prefix/suffix move" optimization.

Why do you say that? 2,3 swap for example would be 995 more loop cycles before it initiates the swap of Dom nodes. Maybe I'll have to test it out, would have thought those extra cycles would make an impact (even if a small one)

**Comment by localvoid** [comment]
> Why do you say that? 2,3 swap for example would be 995 more loop cycles before it initiates the swap of Dom nodes.

Doesn't matter which rows are swapped, it is going to iterate through all nodes.

1. Iterates through nodes with equal keys from left-to-right
2. Iterates through nodes with equal keys from right-to-left
3. Moves node with equal keys from left edge to right edge
4. Moves node with equal keys from right edge to left edge
5. Repeat from step 1 until all nodes are processed

**Comment by robbiespeed** [comment]
My mistake I was thinking the benchmark was somehow finishing after the DOM update was triggered not until idle. The more I think about it the less that makes sense since the layout change doesn't even happen until after the execution loop triggered by the button click.

Thanks @leeoniya for the explanation of why 4,9 was originally changed I was curious about that.
